editor/components/block-drop-zone/index.js

```diff
@@ -53,10 +54,35 @@ function BlockDropZone( { index, isLocked, ...props } ) {
 		}
 	};
 
+	const reorderBlock = ( event, position ) => {
+		if ( index !== undefined && event.dataTransfer ) {
+			try {
+				const { uid, fromIndex, type } = JSON.parse( event.dataTransfer.getData( 'text' ) );
+
+				if ( type !== BLOCK_REORDER ) {
+					return;
+				}
+
+				if ( position.y === 'top' && index > fromIndex ) {
+					props.onDrop( event, uid, index - 1 );
+					return;
```

Comment:
Rather than `return`ing in both of these `if` conditions, why not move the `onDrop` into an else if that's effectively what you're trying to express?

Reply:
Absolutely :-)